### PR TITLE
Xpetra: Add ETI for TpetraMultiVector

### PR DESCRIPTION
**Pull request by csiefer2**
Auto-PR for SHA 0a7ec03

### CODE REVIEW

**Comment by csiefer2** [comment]
I could've sworn I did the Stokhos stuff right....

**Comment by csiefer2** [comment]
@jhux2 I'm off today, so feel free to patch the branch and retest.

File: packages/tpetra/core/cmake/ExplicitInstantiationSupport.cmake [hunk 1]
```diff
@@ -168,6 +173,27 @@ TRIBITS_ETI_GENERATE_MACROS(
...
+    "${${PACKAGE_NAME}_ETI_FIELDS}" "${${PACKAGE_NAME}_ETI_LIBRARYSET}" 
+    "${${PACKAGE_NAME}_ETI_EXCLUDE_SET_GORD_NOT_INT}"
+    list_of_manglings eti_typedefs
+    "TPETRA_INSTANTIATE_GLGN_NO_INT(GO,LO,GO,N)"           TPETRA_ETIMACRO_GLGN_NO_INT)
+
+TRIBITS_ETI_GENERATE_MACROS(
+    "${${PACKAGE_NAME}_ETI_FIELDS}" "${${PACKAGE_NAME}_ETI_LIBRARYSET}" 
+    "${${PACKAGE_NAME}_ETI_EXCLUDE_SET}"
+    list_of_manglings eti_typedefs
+    "TPETRA_INSTANTIATE_LLGN(LO,LO,GO,N)"           TPETRA_ETIMACRO_LLGN)
```
Comment (mhoemmen) [inline packages/tpetra/core/cmake/ExplicitInstantiationSupport.cmake]: Per e-mail discussion, this is a hack that relies on `LO=int`, but I'm OK with that.  We will get rid of GO as a template parameter soon enough, and that will require rewriting all the ETI logic anyway, so we can tolerate some hacks for now.

File: packages/xpetra/src/MultiVector/Xpetra_TpetraMultiVector_def.hpp [hunk 2]
```diff
@@ -358,107 +390,76 @@ namespace Xpetra {
...
+  
+#endif
 
-  template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-  RCP<const MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node > > toXpetra(RCP<const Tpetra::MultiVector< Scalar, LocalOrdinal, GlobalOrdinal, Node > > vec) {
-    if (!vec.is_null())
-      return rcp(new TpetraMultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node >(vec));
-
-    return Teuchos::null;
+  /// \brief Implementation of the assignment operator (operator=);
```
Comment (mhoemmen) [inline packages/xpetra/src/MultiVector/Xpetra_TpetraMultiVector_def.hpp]: `Tpetra::MultiVector::operator=` does not do a deep copy, but `Tpetra::MultiVector::assign` does.  That is, the code looks right, but the documentation isn't quite right.

**Pull request merged**